spec/controllers/http_status_agent_spec.rb

```diff
@@ -36,6 +36,9 @@ def f.get url
 
                        def f.set url, response, time = nil
                          sleep(time/1000) if time
+                         def response.to_hash
```

Comment:
Oy, this is getting ugly.  Not your addition, particularly, but this whole mocking code.

The reason to_hash wasn't working is because the mocked responses are built with `Struct.new(:status, :headers)`, which apparently won't accept to_hash.

It'd be nice if we could at least made a mock response class near the top, something like:

```ruby
class MockResponse < Struct.new(:status, :headers)
  alias_method :to_h, :to_hash
end
```

then replace instances of `Struct.new(:status, :headers).new(` with `MockResponse.new(`.

If you don't want to dive into refactoring this spec, I'm happy to merge your change and then work on it.